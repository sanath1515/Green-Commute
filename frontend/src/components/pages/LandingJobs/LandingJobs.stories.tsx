import React from "react";
import LandingJobs from "./LandingJobs";

export default {
  title: "pages/LandingJobs",
  component: LandingJobs,
};

const Template = (args) => <LandingJobs {...args} />;

export const Primary = Template.bind({});
