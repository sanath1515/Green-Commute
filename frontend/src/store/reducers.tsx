import { optionsType } from "../Constants";

export type greenState = {
  location: string;
  worklocations: string[];
  skills: string[];
  step: string;
  jobId: string;
  searchSkill: string;
  searchLocation: string;
  options: optionsType;
  optionValue: boolean;
  sidebarIndex: string;
};

export const initialGreenState: greenState = {
  location: "",
  worklocations: [],
  skills: ["UI/UX Designer"],
  step: "1",
  jobId: "1",
  searchSkill: "",
  searchLocation: "",
  options: {
    datePosted: [],
    distance: [],
    experienceLevel: [],
    jobType: [],
    transport: [],
  },
  optionValue: true,
  sidebarIndex: "1",
};

type Action = {
  type: string;
  payload: string | string[] | any | boolean;
};

export const greenReducer = (
  state: greenState = initialGreenState,
  action: Action
) => {
  switch (action.type) {
    case "SET_LOCATION": {
      return { ...state, location: action.payload };
    }
    case "SET_SKILLS": {
      return { ...state, skills: action.payload };
    }
    case "SET_STEP": {
      return { ...state, step: action.payload };
    }
    case "SET_WORK_LOCATIONS": {
      return { ...state, worklocations: action.payload };
    }
    case "SET_JOB_ID": {
      return { ...state, jobId: action.payload };
    }
    case "SET_SEARCH_SKILL": {
      return { ...state, searchSkill: action.payload };
    }
    case "SET_SEARCH_LOCATION": {
      return { ...state, searchLocation: action.payload };
    }
    case "SET_OPTIONS": {
      return { ...state, options: action.payload };
    }
    case "SET_OPTIONS_VALUE": {
      return { ...state, optionValue: action.payload };
    }
    case "SET_SIDEBAR_INDEX": {
      return { ...state, sidebarIndex: action.payload };
    }
    default:
      return state;
  }
};
