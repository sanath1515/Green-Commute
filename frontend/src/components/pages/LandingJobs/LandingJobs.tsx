import React from "react";
import SideBar from "../../organisms/SideBar/SideBar";
import TopNavBar from "../../organisms/TopNavBar/TopNavBar";
import { makeStyles } from "@material-ui/core/styles";
import { Grid, Typography } from "@material-ui/core";
import Findjob from "../../organisms/Findjob/Findjob";
import Filters from "../../organisms/Filter/Filters";
import { COMMUTE_ROUTES, jobDataProp } from "../../../Constants";
import JobCard from "../../molecules/JobCard/JobCard";
import useJobsList from "../../customhooks/useJobsList";
import { useSelector } from "react-redux";
import { greenState } from "../../../store/reducers";

const useStyles = makeStyles((theme) => ({
  root: {
    marginLeft: "270px",
    paddingLeft: "60px",
    backgroundColor: "#fafafa",
  },
  grid: { width: "1074px" },
  heading5: {
    fontSize: "20px",
    marginTop: "30px",
    fontWeight: 600,
  },
  heading: {
    marginRight: "722px",
    display: "inline-block",
    marginTop: "30px",
    fontWeight: 600,
  },
  find: {
    marginTop: "15px",
    display: "flex",
    alignItems: "center",
    width: "1050px",
    borderRadius: "6px",
    border: "solid 1px #e3f3f6",
  },
  root1: {
    marginLeft: "260px",
  },
  text: { marginTop: "3px", marginBottom: "20px", color: "#5f7381" },
}));

function LandingJobs() {
  const { jobcards } = useJobsList();
  const location: string = useSelector<
    greenState,
    greenState["searchLocation"]
  >((state) => {
    return state.searchLocation;
  });

  const initiallocation: string = useSelector<
    greenState,
    greenState["location"]
  >((state) => {
    return state.location;
  });

  const skill: string = useSelector<greenState, greenState["searchSkill"]>(
    (state) => {
      return state.searchSkill;
    }
  );

  const classes = useStyles();
  return (
    <>
      <SideBar />
      <div className={classes.root1}>
        <TopNavBar location={initiallocation} src={COMMUTE_ROUTES.userImg} />
      </div>

      <div className={classes.root}>
        <Typography variant="h5" className={classes.heading5}>
          {COMMUTE_ROUTES.findJob}
        </Typography>

        <Findjob ClassName={classes.find} Skill={skill} Location={location} />

        <Typography variant="h5" className={classes.heading}>
          {COMMUTE_ROUTES.recommend}
        </Typography>
        <Filters button="Filter" button1="apply" button2="clear all" />
        <Typography variant="body2" className={classes.text}>
          {COMMUTE_ROUTES.recommendChild}
        </Typography>
        <Grid
          container
          spacing={5}
          justifyContent="flex-start"
          className={classes.grid}
        >
          {jobcards.map((jobs: jobDataProp) => (
            <Grid key={jobs.id} item xs={4}>
              <JobCard
                src={jobs.imageSource}
                days={Math.abs(
                  Math.ceil(
                    (new Date(jobs.dateAdded).getTime() -
                      new Date(Date.now()).getTime()) /
                      (1000 * 3600 * 24)
                  )
                )}
                jobRole={jobs.jobProfile}
                companyName={jobs.companyName}
                location={jobs.city}
                commuteRoute={jobs.commuteRoutes}
              ></JobCard>
            </Grid>
          ))}
        </Grid>
      </div>
    </>
  );
}

export default LandingJobs;
