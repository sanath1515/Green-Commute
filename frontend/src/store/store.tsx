import { createStore } from "redux";
import { greenReducer } from "./reducers";

const store = createStore(greenReducer);

export default store;
