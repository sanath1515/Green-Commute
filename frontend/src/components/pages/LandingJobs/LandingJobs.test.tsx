/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store/store";
import LandingJobs from "./LandingJobs";

describe("LandingJobs Page Test", () => {
  const page = () => {
    render(
      <Router>
        <Provider store={store}>
          <LandingJobs />
        </Provider>
      </Router>
    );
  };

  it("Match Snapshot", () => {
    expect(page).toMatchSnapshot();
  });

  it("To be Required", () => {
    const container = () =>
      render(
        <Router>
          <Provider store={store}>
            <LandingJobs />
          </Provider>
        </Router>
      );
    expect(container).toBeRequired;
  });
});
