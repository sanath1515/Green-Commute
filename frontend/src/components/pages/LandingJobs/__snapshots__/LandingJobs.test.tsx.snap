// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingJobs Page Test Match Snapshot 1`] = `[Function]`;
